enhancements/builds/disable-jenkins-pipeline.md

```diff
@@ -0,0 +1,299 @@
+I want to continue using my Jenkinsfiles
+So that I can continue to use Jenkins for my CI/CD processes
+
+As an OpenShift product manager or SRE
+I want to know which clusters have the JenkinsPipeline strategy enabled
+So that I know which clusters rely on OpenShift's Jenkins integration
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Disabled Build Strategy Behavior
+
+The Jenkins Pipeline build strategy will be disabled by default for new OpenShift clusters.
+When a strategy is disabled, OpenShift will allow Build and BuildConfig objects to reference the disabled strategy.
+However, build controller will immediately fail Build objects that reference a disabled strategy.
+The build's status will clearly indicate that the disabled strategy caused the build to fail.
+The failed build should also record a Kubernetes "Warning" event, indicating that the build failed because the build strategy is disabled.
+
+The code implementing the Jenkins Pipeline build strategy will not be removed.
+Rather, it will be invoked only if the build controller is explicitly configured to enable JenkinsPipeline build strategies via the `BuildJenkinsPipelineStrategy` feature gate.
+
+In keeping with alerting best practices, builds that fail in this fashion should _not_ trigger a Prometheus alert.
+
+#### Feature Gate the Jenkins Pipeline Strategy
+
+The Jenkins Pipeline strategy can be re-enabled through a new feature gate and cluster-wide feature set:
+
+```go
+
+type FeatureSet string
+
+var (
+  ...
+  // DeprecatedJenkinsPipeline enables the deprecated JenkinsPipeline build strategy
+  DeprecatedJenkinsPipeline = "DeprecatedJenkinsPipeline"
+)
+
+...
+
+var FeatureSets = map[FeatureSet]*FeatureGateEnabledDisabled{
+  ...
+  DeprecatedJenkinsPipeline: newDefaultFeatures().
+    with("BuildJenkinsPipelineStrategy").
+    toFeatures(),
+}
+
+```
+
+The operator for openshift-controller-manager will be enhanced to watch the cluster feature gates.
+If the cluster has the `BuildJenkinsPipelineStrategy` feature gate enabled, this gate will be passed to openshift-controller-manager's configuration.
+Openshift-controller-manager will then enable the strategy and allow JenkinsPipline builds to proceed as normal.
+
+#### Bootstrap Behavior
+
+When the `BuildJenkinsPipelineStrategy` feature gate is introduced, the openshift-controller-manager-operator will query all `BuildConfig` objects to see if any utilize the JenkinsPipeline strategy.
+If any BuildConfig uses the JenkinsPipline strategy, the DeprecatedJenkinsPipeline feature set will be turned on.
+
+The bootstrap check will only run if the default feature set is configured for the cluster.
+A status condition will indicate if this bootstrap check has completed.
+Once the bootstrap check is complete, it should not be run again.
+These protections will ensure that we do not overwrite feature sets that are enabled as a day 2 action, or overwrite pre-existing feature sets that allow updgrades.
```

Comment:
when is this bootstrap check performed?  First startup after install, and first startup after upgrade to a version of OCP that has this functionality?